templates/jlink/jlink-builder/jdk-17-buildconfig.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata: 
+  name: jlink-builder-jdk-17
+  labels:
+    app: jlink-builder-jdk-17
+spec:
+  source:
+    dockerfile: |
+      FROM registry.access.redhat.com/ubi9/openjdk-17 AS ubi9-openjdk-17-jmods
```

Comment:
change to `FROM -` , to make explicit that we're going to rely on OpenShift substituting the FROM image with a local reference to an ImageStreamTag. See: https://docs.openshift.com/container-platform/4.15/cicd/builds/build-strategies.html#builds-strategy-docker-from-image_build-strategies for the additional keys we need to add to make that work (`strategy.dockerStrategy.from.kind` and `strategy.dockerStrategy.from.name`, pointing at the UBI9 ImageStream we merged)